CC/src/DistanceComputationTools.cpp

```diff
@@ -1088,8 +1088,7 @@ static bool s_cellFunc_MT_success = true;
 static CCLib::DistanceComputationTools::Cloud2MeshDistanceComputationParams s_params_MT;
 
 //'processTriangles' mechanism (based on bit mask)
-#include <QtCore/QBitArray>
-static std::vector<QBitArray*> s_bitArrayPool_MT;
+static std::vector<std::vector<bool>*> s_bitArrayPool_MT; //TODO: RJ:Do the std::vector<bool> really need to be heap allocated?
```

Comment:
Wow, a QBitArray is much smaller than a std::vector<bool>! 8 times smaller in the ideal case (assuming bool are coded on 8 bits, and I'm not even sure of that). You'll have to find a better structure to replace that...

Reply:
std::vector<bool> should be optimized for space - but you'd have to check with your compiler.

http://www.cplusplus.com/reference/vector/vector-bool/

"The storage is not necessarily an array of bool values, but the library implementation may optimize storage so that each value is stored in a single bit."

Reply:
The loss of performance is even more dramatic than what I exposed in the last message. so std::vector is not an option.

Reply:
Doesn't look like our options are great:

https://www.academia.edu/20674648/Performance_of_C_bit-vector_implementations

`std::bitset` would be great, but the size is fixed at compile-time:

http://www.cplusplus.com/reference/bitset/bitset/

Reply:
I also test with boost::dynamic_bitset, and performance are even worst :(, still investigating...

Reply:
Ok using the right function make boot::dynamic_bitset on parity with QtBitset. The biggest performance bottleneck  with std::vector<bool> is the use of std::fill (see below)